### PR TITLE
ROX-13326: Use UUID columns for roles, permissionsets and simpleaccessscopes in postgres mode.

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

This PR was split out from https://github.com/stackrox/stackrox/pull/3725 . The goal is to isolate the support code from the data migration.

PR stack:
https://github.com/stackrox/stackrox/pull/3802 (server support code)
+- https://github.com/stackrox/stackrox/pull/3725 (data migration)
... +- https://github.com/stackrox/stackrox/pull/3818 (UI code)

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

CI is sufficient.

### CODE REVIEW

File: qa-tests-backend/src/test/groovy/SACTest.groovy [hunk 1]
```diff
@@ -74,7 +74,9 @@ class SACTest extends BaseSpecification {
 
     static final private Integer WAIT_FOR_RISK_RETRIES =
             isRaceBuild() ? 300 : ((Env.mustGetOrchestratorType() == OrchestratorTypes.OPENSHIFT) ? 80 : 50)
-    static final private String DENY_ALL = 'io.stackrox.authz.accessscope.denyall'
+    static final private String DENY_ALL = isPostgresRun() ?
+        "ffffffff-ffff-fff4-f5ff-fffffffffffe" :
+        'io.stackrox.authz.accessscope.denyall'
```
Comment (janisz) [inline qa-tests-backend/src/test/groovy/SACTest.groovy:79]:
```suggestion
    static final private String DENY_ALL = UNRESTRICTED_SCOPE_ID
```

**Review by janisz — APPROVED** [review]

**Comment by rhybrillou** [comment]
The failure in postgres upgrade test is due to the fact that the data migration is in a separate PR.
```
panic: Id is not a valid uuid -- id:"io.stackrox.authz.permissionset.1ee52f65-c548-4b00-8514-0149a19f415d" name:"Continuous Integration Permissions" description:"This permission set was created as part of the migration from the old Role format to the new Role + Permission Set format." resource_to_access:<key:"Detection" value:READ_ACCESS > resource_to_access:<key:"Image" value:READ_WRITE_ACCESS > resource_to_access:<key:"VulnerabilityManagementApprovals" value:READ_WRITE_ACCESS > resource_to_access:<key:"VulnerabilityManagementRequests" value:READ_WRITE_ACCESS > 

goroutine 1 [running]:
github.com/stackrox/rox/pkg/utils.hardPanic({0x1c6ca80?, 0xc00442a7b0})
	github.com/stackrox/rox/pkg/utils/must.go:25 +0xb6
github.com/stackrox/rox/pkg/utils.Should(...)
	github.com/stackrox/rox/pkg/utils/must.go:51
github.com/stackrox/rox/migrator/migrations/n_37_to_n_38_postgres_permission_sets/postgres.insertIntoPermissionSets({0xc000fb7640?, 0x2158ae0?}, 0xc00442a4f8, 0xc000d5c840)
	github.com/stackrox/rox/migrator/migrations/n_37_to_n_38_postgres_permission_sets/postgres/postgres_plugin.go:88 +0x326
github.com/stackrox/rox/migrator/migrations/n_37_to_n_38_postgres_permission_sets/postgres.(*storeImpl).upsert(0xc002e944b8?, {0x2158ae0, 0xc002e94480}, {0xc00349e580, 0xa, 0x14ef646?})
	github.com/stackrox/rox/migrator/migrations/n_37_to_n_38_postgres_permission_sets/postgres/postgres_plugin.go:200 +0x125
github.com/stackrox/rox/migrator/migrations/n_37_to_n_38_postgres_permission_sets/postgres.(*storeImpl).UpsertMany.func1()
	github.com/stackrox/rox/migrator/migrations/n_37_to_n_38_postgres_permission_sets/postgres/postgres_plugin.go:236 +0xe9
github.com/stackrox/rox/pkg/postgres/pgutils.Retry.func1()
	github.com/stackrox/rox/pkg/postgres/pgutils/retry.go:22 +0x1b
```
The migration code in https://github.com/stackrox/stackrox/pull/3725 takes care of converting the data to UUID values.

**Pull request merged**